### PR TITLE
support to send attributes set in the script tag

### PR DESCRIPTION
**Pull request by javisantana**
this sends a properties starting with `tb_` in as json props.

```
<script tb_test="ABC" tb_foo="1".... ></script>
```

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have update the [CHANGELOG](https://github.com/tinybirdco/web-analytics-starter-kit/blob/main/CHANGELOG.md)
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: middleware/src/index.js [hunk 1]
```diff
@@ -2,13 +2,21 @@
...
 
     let proxy, token, host;
     if (document.currentScript) {
         host = document.currentScript.getAttribute('data-host');
         proxy = document.currentScript.getAttribute('data-proxy');
         token = document.currentScript.getAttribute('data-token');
         DATASOURCE = document.currentScript.getAttribute('data-datasource') || DATASOURCE;
+       
+        for (const attr of  document.currentScript.attributes) {
+           if( attr.name.startsWith('tb_') {
```
Comment (ricardotinybird) [inline middleware/src/index.js:15]: I think it is missing a ')' here

File: middleware/src/index.js [hunk 2]
```diff
@@ -88,6 +96,8 @@
         }
 
         payload = _maskSuspiciousAttributes(payload);
+        
+        payload = {...payloadglobalAttributes, payload}
```
Comment (dmytro-tinybird) [inline middleware/src/index.js:100]: should be {...payload, ...globalAttributes}

File: middleware/src/index.js [hunk 3]
```diff
@@ -96,8 +96,8 @@
         }
 
         payload = _maskSuspiciousAttributes(payload);
-        
-        payload = {...payloadglobalAttributes, payload}
+        payload = Object.assign({}, JSON.parse(payload), globalAttributes);
```
Comment (javisantana) [inline middleware/src/index.js:99]: why this parse here?

Comment (ricardotinybird) [reply middleware/src/index.js:99]: because _maskSuspiciousAttributes returns String type

Comment (javisantana) [reply middleware/src/index.js:99]: then why don't we move the globalAttributes before calling it?

**Review by ricardotinybird — DISMISSED** [review]

**Review by alejandromav — APPROVED** [review]

**Pull request merged**